src/coinjoin/interfaces.cpp

```diff
@@ -69,19 +69,19 @@ class CoinJoinLoaderImpl : public interfaces::CoinJoin::Loader
 
     void AddWallet(CWallet& wallet) override
     {
-        m_walletman.Add(wallet);
+        WITH_LOCK(m_walletman.cs_wallet_manager_map, m_walletman.Add(wallet));
     }
     void RemoveWallet(const std::string& name) override
     {
-        m_walletman.Remove(name);
+        WITH_LOCK(m_walletman.cs_wallet_manager_map, m_walletman.Remove(name));
     }
     void FlushWallet(const std::string& name) override
     {
-        m_walletman.Flush(name);
+        WITH_LOCK(m_walletman.cs_wallet_manager_map, m_walletman.Flush(name));
     }
     std::unique_ptr<interfaces::CoinJoin::Client> GetClient(const std::string& name) override
     {
-        auto clientman = m_walletman.Get(name);
+        auto clientman = WITH_LOCK(m_walletman.cs_wallet_manager_map, return m_walletman.Get(name));
```

Comment:
this should just be guarded inside, there's really no point the way that it is here locking externally and then immediately dropping the lock

Reply:
Ideally we shouldn't be accessing it directly at all but instead through the interface, but functions like `DoAutomaticDenominating` and `GetStatuses` are not implemented as part of that interface and as explained in my comment earlier ([source](https://github.com/dashpay/dash/pull/6192#discussion_r1712971813)), moving it inside causes a small problem because every time we `Add`, `Remove` or `Flush`, we also re-init CoinJoin.

During re-init, it calls `Get` to find the manager associated with the wallet to stop (and conditionally restart) mixing, which means, if we move the `LOCK` to be internal to the functions, it will be held twice but if we keep the current external approach, it only needs to be held once (by the interface that calls `Add`, `Remove` or `Flush`).

Reply:
Have moved to using a `mutable RecursiveMutex`, resolved in latest push!

Reply:
pls consider https://github.com/dashpay/dash/pull/6192#discussion_r1713690690 - no recursive mutexes, no direct access to the internal map from the outside via `raw()`

Reply:
Cherry-picked, resolved in latest push!